packages/integrations/solid/README.md

```diff
@@ -98,6 +98,73 @@ export default defineConfig({
 });
 ```
 
+## Rendering strategy
+
+Hydrating SolidJS components are automatically wrapped in Suspense boundaries and rendered on the server using the [renderToStringAsync](https://www.solidjs.com/docs/latest/api#rendertostringasync) function. This means that lazy components will be resolved and rendered on the server. For example the following source files:
```

Comment:
In general, we prefer not to have multiple code samples just thrown at the reader with no commentary, and then at the very end ask them to make a connection about what they saw. We do try to guide the reader by priming them for each component they're about to see.

Here, you're not guiding the reader at all as to what to even look for, so they have to read every. single. thing. super closely to see what's important and what is actually happening. Is the name of the file important? (turns out it is!) Is the fact that the name `Hello` in the function important? (Turns out it's not!) 

These should really be telling the reader what they are *going to see*, so that they already know the _what_ before they get there, and they can just focus on the _how_.

So, can you try to do that here, something like (and I'm not really familiar with Solid, so this is just an example of how to guide the reader. It's your job to make these statements accurate!)

Before you do that for this first example, though, please read my comment for the second example. I suspect something similar is happening here, where there might be more code samples than are needed to simply convey that "you can lazily import other components, and they'll be available." I found this super long and complicated to get that idea across, and no priming for what to look for. The second example is a bit more concrete, and I think whatever strategy you end up using there will help this example.

Reply:
Wow, thanks for deep feedback. :pray: 

I cut down the examples a lot and decided to focus on `createResource`, as that seems easier to explain and understand.

I left one line in about `lazy`. Maybe it's less critical to explain to the end user, but I think this PR should fix the lazy-related bug #6912, so I feel like it should at least get a small mention.

Addressed in: https://github.com/withastro/astro/pull/6791/commits/5a810a9b73463d93b3b7eeee92ea1dd1ca9f7111